test.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+
+from DIRACbenchmark import *
+import pytest
+
+
+@pytest.mark.parametrize(
+    "copies, iterations, extraIteration",
+    [
+        ("single", 1, False),
+        ("single", 2, True),
+        ("wholenode", 1, False),
+        ("wholenode", 2, True),
+        ("jobslot", 1, False),
+        ("jobslot", 2, True),
+        (2, 1, False),
+        (3, 2, True),
+    ],
+)
+def testDIRACbenchmark(copies, iterations, extraIteration):
+    k = 0
+
+    if copies is None or copies == "single":
+        result = singleDiracBenchmark(iterations=iterations)["NORM"]
+        result2 = singleDiracBenchmark(iterations=iterations)["NORM"]
+        result3 = singleDiracBenchmark(iterations=iterations + 1)["NORM"]
+
+        if result2 <= result:
+            assert result - result2 <= result2 * 20 / 100
```

Comment:
Use the absolute value: `abs(result - result2)`.
Example: 
- result: 20
- result2:30

with your test it would give:
```python
assert -10 <= 6 #pass while it should not
```

Reply:
I avoided using the absolute value just to decide on which variable to use in the right hand side. In this case result is less than result2 so I assume it will check the else part instead of the if?

Reply:
Alright, so actually you can replace the if else block by `abs`:

```python
if result2 <= result:
    assert result - result2 <= result2 * 0.2
else:
    assert result2 - result <= result * 0.2
```

should be equivalent to:

```python
assert abs(result2-result) <= result * 0.2
```

Since normally `result` and `result2` should be approximately the same, you can use either `result *0.2` or `result2 * 0.2`.